### PR TITLE
BUG: Expose bug with scraping

### PR DESCRIPTION
**Pull request by larsoner**
Just a companion PR to https://github.com/pyvista/pyvistaqt/pull/317. @banesullivan feel free to push fixes here (or to my `pyvistaqt` branch) if you think `pyvista` does indeed need to be updated to fix the bug. It should show the failure here same as it already does in https://github.com/pyvista/pyvistaqt/pull/317

Notably the `pyvista` test passes, presumably because of the `off_screen=True` or something? I guess multiple renderers there can be current, whereas in pyvistaqt they (rightly) cannot...

### CODE REVIEW

**Comment by banesullivan** [comment]
The tests here seem to be passing... 😕

**Comment by larsoner** [comment]
> The tests here seem to be passing... 😕

Argh yes:
```
SKIPPED [2] tests/test_plotting.py:963: could not import 'sphinx_gallery': No module named 'sphinx_gallery'
```

This skipping has been fixed now, it's properly failing here the same way in [this PR's run](https://github.com/pyvista/pyvista/actions/runs/4233290667/jobs/7354061127#step:10:43) as it is in [this run](https://github.com/pyvista/pyvistaqt/actions/runs/4196914929/jobs/7278542082#step:10:94) from https://github.com/pyvista/pyvistaqt/pull/317 😄 
```
=================================== FAILURES ===================================
_______________________ test_sphinx_gallery_scraping[2] ________________________
tests/test_plotting.py:991: in test_sphinx_gallery_scraping
    scraper(block, block_vars, gallery_conf)
/opt/hostedtoolcache/Python/3.10.10/x64/lib/python3.10/site-packages/pyvista/utilities/sphinx_gallery.py:60: in __call__
    plotter.screenshot(fname)
/opt/hostedtoolcache/Python/3.10.10/x64/lib/python3.10/site-packages/pyvista/plotting/plotting.py:5609: in screenshot
    return self._save_image(self.image, filename, return_img)
pyvistaqt/rwi.py:426: in __getattr__
    raise AttributeError(self.__class__.__name__ +
E   AttributeError: BackgroundPlotter has no attribute named image
```

**Comment by larsoner** [comment]
@banesullivan any ideas here?

**Comment by banesullivan** [comment]
Added to my list of tasks for the week, will jump on this

File: tests/test_scraper.py [hunk 1]
```diff
@@ -12,29 +12,35 @@
...
-    img_fname = op.join(src_dir, 'auto_examples', 'images', 'sg_img.png')
+    img_fnames = [
+        op.join(src_dir, 'auto_examples', 'images', f'sg_img_{n}.png') for n in range(n_win)
+    ]
     gallery_conf = {"src_dir": src_dir, "builder_name": "html"}
     target_file = op.join(src_dir, 'auto_examples', 'sg.py')
     block = None
     block_vars = dict(
-        image_path_iterator=(img for img in [img_fname]),
+        image_path_iterator=(img for img in img_fnames),
```
Comment (adeak) [inline tests/test_scraper.py:31]: Perhaps
```suggestion
        image_path_iterator=iter(img_fnames),
```
?

Looking at this, I'm not sure why [in the scraper](https://github.com/pyvista/pyvista/blob/2096a0a4d47b33112dc04ef2184ec17b9cdf5bb1/pyvista/utilities/sphinx_gallery.py#L52-L61) we don't just do `image_path_iterator = iter(block_vars['image_paths'])`, or even better, just `zip(figures.items(), image_paths)`. Irrelevant for this PR obviously.

Comment (adeak) [reply tests/test_scraper.py:31]: Unless the idea is that multiple calls to the scraper keep using further items from the paths. In that case calling `iter()` on the input would probably still work to allow sequences to be passed in initially. Not sure.

File: pyvista/utilities/sphinx_gallery.py [hunk 2]
```diff
@@ -57,6 +57,7 @@ def __call__(self, block, block_vars, gallery_conf):
                 # move gif to fname
                 shutil.move(plotter._gif_filename, fname)
             else:
+                # plotter.ren_win.MakeCurrent()
```
Comment (akaszynski) [inline pyvista/utilities/sphinx_gallery.py:60]:
```suggestion
```

File: .github/workflows/integration-tests.yml [hunk 3]
```diff
@@ -70,12 +70,17 @@ jobs:
       - name: Install Qt and testing dependencies
         run: pip install PyQt6 pytest pytest-qt pytest-cov ipython
       - name: Clone PyVistaQt
-        run: git clone --depth=1 https://github.com/pyvista/pyvistaqt.git --branch main --single-branch
+        # TODO: REVERT THIS BEFORE MERGE!
+        # run: git clone --depth=1 https://github.com/pyvista/pyvistaqt.git --branch main --single-branch
+        run: git clone --depth=1 https://github.com/larsoner/pyvistaqt.git --branch bug --single-branch
```
Comment (larsoner) [inline .github/workflows/integration-tests.yml:75]: Thanks for looking into this -- just don't forget to revert this before merge!

**Comment by akaszynski** [comment]
Before changing back to `main`, https://github.com/pyvista/pyvista/actions/runs/4510012038/jobs/7940454091 shows that this branch indeed resolves the issue.

**Review by akaszynski — APPROVED** [review]
LGTM.

Should resolve https://github.com/pyvista/pyvistaqt/issues/324 as well.

**Review by larsoner — COMMENTED** [review]
I am not allowed to click "approve" because it's my PR, but I approve. Thanks for the fix!

**Review by adeak — APPROVED** [review]
Approving on behalf of @larsoner 😄

**Pull request merged**